usaspending_api/etl/es_etl_helpers.py

```diff
@@ -452,43 +446,47 @@ def csv_chunk_gen(filename, chunksize, job_id, load_type):
         # and this is needed for performance
         # ES helper will pop any "meta" fields like "routing" from provided data dict and use them in the action
         file_df["routing"] = file_df[settings.ES_ROUTING_FIELD]
+
+        # Explicitly setting the ES _id field to match the postgres PK value allows
+        # bulk index operations to be upserts without creating duplicate documents
+        file_df["_id"] = file_df[f"{'award' if load_type == 'awards' else 'transaction'}_id"]
         yield file_df.to_dict(orient="records")
 
 
 def es_data_loader(client, fetch_jobs, done_jobs, config):
     if config["create_new_index"]:
         # ensure template for index is present and the latest version
-        call_command("es_configure", "--template-only", "--load_type={}".format(config["load_type"]))
+        call_command("es_configure", "--template-only", f"--load-type={config['load_type']}")
     while True:
         if not done_jobs.empty():
             job = done_jobs.get_nowait()
             if job.name is None:
                 break
 
-            printf({"msg": "Starting new job", "job": job.name, "f": "ES Ingest"})
+            printf({"msg": "Starting new job", "job": job.name, "f": "ES Index"})
             post_to_elasticsearch(client, job, config)
             if os.path.exists(job.csv):
                 os.remove(job.csv)
         else:
-            printf({"msg": "No Job. Sleeping 45s", "f": "ES Ingest"})
-            sleep(45)
+            printf({"msg": f"No Job. Sleeping {config['ingest_wait']}s", "f": "ES Index"})
+            sleep(int(config["ingest_wait"]))
 
-    printf({"msg": "Completed Elasticsearch data load", "f": "ES Ingest"})
+    printf({"msg": "Completed Elasticsearch data load", "f": "ES Index"})
     return
 
 
 def streaming_post_to_es(client, chunk, index_name: str, type: str, job_id=None):
     success, failed = 0, 0
     try:
-        for ok, item in helpers.streaming_bulk(client, chunk, index=index_name):
+        for ok, item in helpers.parallel_bulk(client, chunk, index=index_name):
```

Comment:
Some minor quibbles about the name of our wrapper function here. What makes it "streaming"? To me streaming would imply that as soon as data is changing some event handling is occurring to stream the event to its final destination. 
Other options for perhaps more accurate is batching, or microbatching. Probably batching is still most accurate since microbatching is a way to pile-up streaming events into small batches. And then use of parallel is a good qualifier in the name if executing batches in parallel.

Reply:
I can understand the ambiguity. From the elasticsearch library method perspective, `streaming_bulk()` and `parallel_bulk()` are streaming a batch load. The chunk size is 250K, but each load iteration is only 500 records (the functions are iterators/generators, which is why they are looping in the for statement).